src/ToricVarieties/NormalToricVarieties/attributes.jl

```diff
@@ -317,6 +317,106 @@ end
 export toric_ideal
 
 
+@doc Markdown.doc"""
+    set_coordinate_names_of_torus(v::AbstractNormalToricVariety, coordinate_names::Vector{String})
+
+Allows to set the names of the coordinates of the torus.
+"""
+function set_coordinate_names_of_torus(v::AbstractNormalToricVariety, coordinate_names::Vector{String})
+    if length(coordinate_names) != ambient_dim(v)
+        throw(ArgumentError("The provided list of coordinate names must match the ambient dimension of the fan."))
+    end
+    set_attribute!(v, :coordinate_names_of_torus, coordinate_names)
+end
+export set_coordinate_names_of_torus
+
+
+@doc Markdown.doc"""
+    coordinate_names_of_torus(v::AbstractNormalToricVariety)
+
+This method returns the names of the coordinates of the torus of
+the normal toric variety `v`. The default is `x1,...,xn`.
+"""
+coordinate_names_of_torus(v::AbstractNormalToricVariety) = get_attribute!(v, :coordinate_names_of_torus, ["x$(i)" for i in 1:ambient_dim(v)])
+export coordinate_names_of_torus
+
+
+@doc Markdown.doc"""
+    coordinate_ring_of_torus(v::AbstractNormalToricVariety, R::MPolyRing)
+
+Computes the coordinate ring of the torus of the normal toric variety `v`
+in the given polynomial ring `R`.
+"""
+function coordinate_ring_of_torus(v::AbstractNormalToricVariety, R::MPolyRing)
+    if length(gens(R)) < 2 * length(coordinate_names_of_torus(v))
```

Comment:
A matter of taste but also a micro optimization: if you store `length(coordinate_names_of_torus(v))` in a local var, then substitute below, the code IMHO gets easier to read.
```suggestion
    n = length(coordinate_names_of_torus(v))
    if length(gens(R)) < 2 * n
```

Reply:
Thx. Adjusted.